apps/challenges/views.py

```diff
@@ -2969,6 +2970,83 @@ def get_worker_logs(request, challenge_pk):
     return Response(response_data, status=status.HTTP_200_OK)
 
 
+@api_view(["PUT"])
+@throttle_classes([UserRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((JWTAuthentication, ExpiringTokenAuthentication))
+def scale_resources_by_challenge_pk(request, challenge_pk):
+    """
+    The function called by a host to update the resources used by their challenge.
+
+    Calls the scale_resources method. Before calling, checks if the caller hosts the challenge and provided valid CPU
+    unit counts and memory sizes (MB).
+
+    Arguments:
+        request {HttpRequest} -- The request object
+        challenge_pk {int} -- The challenge pk for which its workers' resources will be updated
+
+    Returns:
+        Response object -- Response object with appropriate response code (200/400/403/404)
+    """
+    if not is_user_a_host_of_challenge(request.user, challenge_pk):
+        response_data = {
+            "error": "Sorry, you are not authorized for access worker operations."
+        }
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    if request.data.get("cpu_units") is None:
+        response_data = {
+            "error": "Number of CPU units is missing from request."
+        }
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    if request.data.get("memory") is None:
+        response_data = {
+            "error": "Amount of memory is missing from request."
+        }
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    challenge = get_challenge_model(challenge_pk)
+    cpu_units = request.data["cpu_units"]
+    memory = request.data["memory"]
+
+    try:
+        cpu_units = int(cpu_units)
+        memory = int(memory)
+    except Exception:
+        response_data = {
+            "error": "Enter integer numbers for cores and memory."
+        }
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Is there a cleaner way to write this check? A separate try except just for parsing seems bad idea

Reply:
This is the cleanest way to convert the entries into ints, as is shown [here](https://stackoverflow.com/questions/736043/checking-if-a-string-can-be-converted-to-float-in-python).